src/core/DatetimeClasses.ts

```diff
@@ -0,0 +1,475 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    getOffsetSecondsFromTimezoneString,
+    getTimezoneOffsetFromSeconds
+} from '../util/DatetimeUtil';
+
+/**
+ * ### Local time object
+ * * Represents time in day without timezone.
+ */
+export class HzLocalTime {
+    private static readonly timeStringRegex = /(\d\d):(\d\d):(\d\d)(\.\d+)?/;
+
+    private constructor(readonly hour: number, readonly minute: number, readonly second: number, readonly nano: number) {
+        if (!Number.isInteger(hour) || !Number.isInteger(minute) || !Number.isInteger(second) || !Number.isInteger(nano)) {
+            throw new RangeError('Illegal arguments given to HzLocalTime. All arguments must be integers.');
+        }
+        if (!(hour >= 0 && hour <= 23)) {
+            throw new RangeError('Hour-of-day must be between 0-23');
+        }
+        if (!(minute >= 0 && minute <= 59)) {
+            throw new RangeError('Minute-of-hour must be between 0-59');
+        }
+        if (!(second >= 0 && second <= 59)) {
+            throw new RangeError('Second-of-minute must be between 0-59');
+        }
+        if (!(nano >= 0 && nano <= 999_999_999)) {
+            throw new RangeError('Nano-of-second must be between 0-999_999_999');
+        }
+    }
+
+    /**
+     * @internal
+     * Static constructor that is used by constructor function
+     */
+    static _new(hour: number, minute: number, second: number, nano: number) {
+        return new HzLocalTime(hour, minute, second, nano);
+    }
+
+    /**
+     * Constructs a new {@link HzLocalTime} object from timeString.
+     * @param timeString A string in the form HH:mm:ss.SSS, where the last part represents nanoseconds and optional.
+     * At most 9 digits allowed for nanosecond value. If more than 9 digits are given, the first 9 of them are used.
+     * @throws RangeError if invalid timeString is given
+     */
+    static fromString(timeString: string): HzLocalTime {
+        if (typeof timeString !== 'string') {
+            throw new RangeError('String expected.');
+        }
+        const match = timeString.match(HzLocalTime.timeStringRegex);
+        if (!match) {
+            throw new RangeError('Illegal time string. Expected a string in HH:mm:ss.SSS format');
+        }
+        const hours = +match[1];
+        const minutes = +match[2];
+        const seconds = +match[3];
+
+        let nano = 0;
+        if (match[4] !== undefined) { // nano second included
+            let nanoStr = match[4].substring(1, 10); // does not include first dot
+            nanoStr = nanoStr.padEnd(9, '0');
+            nano = +nanoStr;
+        }
+
+        if (isNaN(hours) || isNaN(minutes) || isNaN(seconds) || isNaN(nano)) {
+            throw new RangeError('Illegal time string. Expected a string in HH:mm:ss.SSS format');
+        }
+        return new HzLocalTime(hours, minutes, seconds, nano);
+    }
+
```

Comment:
should we provide asDate function that converts this to native Date object with zero values on year mont day etc.?

Reply:
Maybe similarly a function like fromDate?

Reply:
For year 0-99 maps to 1900-1999 in a Date constructor. So some values won't be able to be represented as `asDate` result.
![image](https://user-images.githubusercontent.com/32355782/125651516-b26d7de4-5416-432f-8bea-d21e9862bf99.png)


I think it is not very hard to use the constructor since there are 3 or 4 fields

Reply:
Still I am not sure though

Reply:
Anyways, I think we can go over this later and add it if necessary in the future